database/migrations/2025_08_28_015257_update_resource_table_change_resource_type_datatype.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+use App\Enums\ResourceType;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class () extends Migration {
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('resources', function (Blueprint $table) {
+            $allowed = ResourceType::toArray();
+            $enumList = "'" . implode("','", $allowed) . "'";
+            if (Schema::hasColumn('resources', 'resource_type')) {
+                DB::table('resources')->whereNotIn('resource_type', $allowed)->update(['resource_type' => null]);
+                DB::statement("ALTER TABLE `resources` MODIFY COLUMN `resource_type` ENUM({$enumList}) NULL AFTER `task_id`");
+            } else {
```

Comment:
sure na ni nga logic? kay i think outside ni sa else.
kung i sulod ni og else. ma change ra into enum if ma false ang hasColumn
i think imo process nga gi huna huna
is
if naay problematic data. fix una.
kung wala proceed.
pag ma fix/or walay problematic data i change na into enum